src/draft/NSG/Agent/LegalEntity/NonListedCompany/Establishment/Write.py

```diff
@@ -0,0 +1,1633 @@
+        "more granular than level 1.",
+        example="Lapland",
+        max_length=40,
+    )
+
+
+class ManagingDirectors(CamelCaseModel):
+    role: ManagingDirectorRole = Field(
+        ...,
+        title="List of director roles",
+        description="The role of the director",
+        example=ManagingDirectorRole.DIRECTOR,
+    )
+    given_name: str = Field(
+        ...,
+        title="Given name",
+        description="The first name that the person is being called by",
+        example="Mary",
+        max_length=250,
+    )
+    middle_names: str = Field(
+        ...,
+        title="Middle name",
+        description="All the middle names of the person",
+        example="Juliet Olive",
+        max_length=250,
+    )
+    last_name: str = Field(
+        ...,
+        title="Last name",
+        description="The person's current family name",
+        example="Deo",
+        max_length=250,
+    )
+    date_of_birth: date = Field(
+        ...,
+        title="Date of birth",
+        description="The birth day of a person",
+        example=date(1976, 4, 16),
+    )
+    nationality: ISO_3166_1_Alpha_3 = Field(
+        ...,
+        title="Nationality",
+        description="The nationality of a person",
+        example=ISO_3166_1_Alpha_3.USA,
+    )
+
+
+class BoardMembers(CamelCaseModel):
+    role: BoardMemberRole = Field(
+        ...,
+        title="Role",
+        description="The role of the person in the board",
+        example=BoardMemberRole.MEMBER,
+    )
+    given_name: str = Field(
+        ...,
+        title="Given name",
+        description="The first name that the person is being called by",
+        example="Mary",
+        max_length=250,
+    )
+    middle_names: str = Field(
+        ...,
+        title="Middle names",
+        description="All the middle names of the person",
+        example="Juliet Olive",
+        max_length=250,
+    )
+    last_name: str = Field(
+        ...,
+        title="Last name",
+        description="The person's current family name",
+        example="Deo",
+        max_length=250,
+    )
+    date_of_birth: date = Field(
+        ...,
+        title="Date of birth",
+        description="The birth day of a person",
+        example=date(1976, 4, 16),
+    )
+    nationality: ISO_3166_1_Alpha_3 = Field(
+        ...,
+        title="Nationality",
+        description="The nationality of a person",
+        example=ISO_3166_1_Alpha_3.USA,
+    )
+
+
+class AuditorDetails(CamelCaseModel):
+    company_name: Optional[str] = Field(
+        None,
+        title="Company name",
+        description="The name of the auditor company if exists",
+        example="Auditor company X",
+        max_length=250,
+    )
+    national_identifier: Optional[str] = Field(
+        None,
+        title="National identifier",
+        description="The national identifier of the non-listed company issued by the trade register",
```

Comment:
@hsuomi: Care to check through this description? I'd somehow expect this is the national identifier of the auditor and I'd guess the auditor could be a listed company and not necessarily a non-listed one. Currently exactly as in the spreadsheet.

Reply:
I checked and updated spreadsheet accordingly.